radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/ParticleGroup.java

```diff
@@ -0,0 +1,119 @@
+package com.radixdlt.client.core.atoms;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.client.core.atoms.particles.Particle;
+import com.radixdlt.client.core.atoms.particles.Spin;
+import com.radixdlt.client.core.atoms.particles.SpunParticle;
+import org.radix.serialization2.DsonOutput;
+import org.radix.serialization2.SerializerId2;
+import org.radix.serialization2.client.SerializableObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * A group of particles representing one intent, e.g. a transfer.
+ * <p>
+ * * @author flotothemoon
+ */
+@SerializerId2("PARTICLEGROUP")
+public class ParticleGroup extends SerializableObject {
+	@JsonProperty("particles")
+	@DsonOutput(DsonOutput.Output.ALL)
+	private final ImmutableList<SpunParticle> particles;
+
+	private ParticleGroup() {
+		this.particles = ImmutableList.of();
+	}
+
+	public ParticleGroup(List<SpunParticle> particles) {
```

Comment:
This could be a Collection or even Iterable.

Reply:
I thought I did that already. Huh.